### ISSUE TITLE
Support to validate against a schema

### ISSUE THREAD

**Issue by dNitro** [adrienverge/yamllint#37]
[jsonlint](https://github.com/zaach/jsonlint) has a feature to check against an existing schema file through the following cli options:
```bash
-V, --validate    a JSON schema to use for validation
-e, --environment which specification of JSON Schema the validation file uses [json-schema-draft-03]
```
It would be great if *yamllint* also could provide such feature.

**Comment by adrienverge** [comment]
Hi @dNitro, the idea looks great but this is out of scope of a linter. Another tool could be greated for that.

**Comment by dNitro** [comment]
@adrienverge, Though of scope collision, but i think that's why they propose as an option. Editors do all types of these operations under **lint** tag. I don't know too much python but using tools like [http://pyyaml.org/wiki/PyYAML](http://pyyaml.org/wiki/PyYAML) and [jsonschema](https://github.com/Julian/jsonschema), it should be practicable (like this SO answer implementation: http://stackoverflow.com/a/22231372/5048383 ). Anyway its a nice option to have.

**Comment by adrienverge** [comment]
I see a real difference between a *linter* and a *schema validator*. So in my opinion, jsonlint is both, they made the choice to be a kind of toolbox.

Have you looked at https://github.com/Grokzen/pykwalify? It seems to do the job.

**Comment by dNitro** [comment]
Of course they are good options as long as they don't cross red lines ;) yes, pykwalify does the job. Thanks. yamllint is a nice linter. keep up the good work man.

**Comment by ssbarnea** [comment]
Any chance to reconsider the lack of schema support? I think that schema validation woudl bring a huge amount of value to yamllint, especially as tools like ansible, travis,... do already have published schemas, so there is no need to implement any schema yourself.

**Comment by adrienverge** [comment]
I agree such a feature brings a huge value, but many other features also would.
It's just out of the scope of linting, I think.

**Comment by ssbarnea** [comment]
I think we may have chance to get this feature added to https://github.com/redhat-developer/yaml-language-server/issues/106
